great_expectations/core/partitioners.py

```diff
@@ -0,0 +1,64 @@
+from typing import List, Literal, Union
+
+from great_expectations.compatibility import pydantic
+
+
+class PartitionerYear(pydantic.BaseModel):
+    column_name: str
+    method_name: Literal["partition_on_year"] = "partition_on_year"
```

Comment:
Non-blocking, but a couplethings on these method names:
* Can we remove them from the constructor's signature altogether? Do pydantic v1 fields support `init=False`?
* Or should it be a ClassVar?
* Even though it's a literal, it's still a magic string, in a followup, should we connect these with the backend via an enum or something along those lines?